server/app/core/tts_service.py

```
@@ -0,0 +1,67 @@
+    ) -> bytes:
+        provider = self.providers.get(agent_config.tts_type)
+        if not provider:
+            raise ValueError(f"Unsupported TTS provider: {agent_config.tts_type}")
+        
+        try:
+            return await provider.generate_speech(
+                text,
+                agent_config
+            )
+        except Exception as e:
+            raise Exception(f"Failed to generate speech: {str(e)}")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)